lib.go

```diff
@@ -712,6 +712,11 @@ func readSheetsFromZipFile(f *zip.File, file *File, sheetXMLMap map[string]strin
 	if err != nil {
 		return nil, nil, err
 	}
+	defer func() {
+		if err = rc.Close(); err != nil {
+			return nil, nil, err
```

Comment:
You can't return the outer functions results from a deferred function like this.  You instead could try and do the close at the return points below.